Framework/Geometry/src/Instrument/InstrumentDefinitionParser.cpp

```diff
@@ -1319,6 +1320,148 @@ void InstrumentDefinitionParser::appendLeaf(Geometry::ICompAssembly *parent,
           "Duplicate detector ID found when adding RectangularDetector " +
           name + " in XML instrument file" + filename);
     }
+  } else if (category.compare("StructuredDetector") == 0 ||
```

Comment:
The first three of these can be bundled into a single case-insensitive comparsion. I would go further and use a `boost::regex` to match all three with a simple pattern.